x-pack/plugins/event_log/server/event_log_client.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable } from 'rxjs';
+import { ClusterClient, SavedObjectsClientContract } from 'src/core/server';
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { EsContext } from './es';
+import { IEventLogClient, IEvent } from './types';
+export type PluginClusterClient = Pick<ClusterClient, 'callAsInternalUser' | 'asScoped'>;
+export type AdminClusterClient$ = Observable<PluginClusterClient>;
+
+interface EventLogServiceCtorParams {
+  esContext: EsContext;
+  savedObjectsClient: SavedObjectsClientContract;
+}
+
+const optionalDateFieldSchema = schema.maybe(
+  schema.string({
+    validate(value) {
+      if (isNaN(Date.parse(value))) {
+        return 'Invalid Date';
+      }
+    },
+  })
+);
+
+export const findOptionsSchema = schema.object({
```

Comment:
We probably should support a few more things like sorting and searching at the API level to support the searchbox and table in the UI?

Reply:
I was going to leave that for when those features were being built, keep the PR smaller 🤷‍♂ 

But sure, I've now added sorting.
I'm not sure how we'd want to support searching with a searchbox so I'll leave that for the PR that actually specifies that.

Reply:
Fair enough, thanks for adding the sorting 👍